dbschema/product-progress.esdl

```diff
@@ -0,0 +1,19 @@
+module default {
+  type ProductProgress extending Mixin::Timestamped {
+    product: Product;
+    report: PeriodicReport;
+    variant: VariantProgress::Variant;
+    multi steps: StepProgress;
+  }
+  type StepProgress extending Mixin::Timestamped {
+    step: Product::Step;
+    completed: float32;
+  }
+}
+
+module VariantProgress {
+  scalar type Variant extending enum<
```

Comment:
Typically I've kept variants loose strings, so that they can be changed easily. But since this isn't polymorphic, like PromptVariantResponse, and there's only two, and I doubt this will change I'm fine keeping it.